src/onnx/parse_mean.cpp

```diff
@@ -24,17 +28,29 @@ struct parse_mean : op_parser<parse_mean>
         auto divisor = info.add_literal(
             migraphx::literal{migraphx::shape{args[0]->get_shape().type()}, {num_data}});
 
-        // TODO: Only divide when using floating-point
-        return std::accumulate(args.begin() + 1,
-                               args.end(),
-                               info.add_broadcastable_binary_op("div", args[0], divisor),
-                               [&](auto mean, auto data_i) {
-                                   // Pre-divide each tensor element-wise by n to reduce risk of
-                                   // overflow during summation
-                                   auto div =
-                                       info.add_broadcastable_binary_op("div", data_i, divisor);
-                                   return info.add_broadcastable_binary_op("add", mean, div);
-                               });
+        if(contains(float_types, args[0]->get_shape().type()))
```

Comment:
I wonder if we should add a method to the shape class `is_float_type()` instead of creating a set with the float types. Just because it might be useful elsewhere in the future. If others agree, we can make this a TODO and open a separate issue.

Reply:
Yea, I agree. We could use the visitor for this to check the type using type_traits instead of having raw lists. Thats what I did in the mlir-c branch to check for signedness:

https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/mlir-c/src/include/migraphx/shape.hpp#L171

We should open a seperate issue for this.